gremlin-dotnet/src/Gremlin.Net/Driver/GremlinClient.cs

```diff
@@ -58,14 +58,15 @@ public class GremlinClient : IGremlinClient
         ///     A delegate that will be invoked with the <see cref="ClientWebSocketOptions" />
         ///     object used to configure WebSocket connections.
         /// </param>
+        /// <param name="sessionId">The session Id if Gremlin Client in session mode, defaults to null as session-less Client.</param>
         public GremlinClient(GremlinServer gremlinServer, GraphSONReader graphSONReader = null,
             GraphSONWriter graphSONWriter = null, string mimeType = null,
-            Action<ClientWebSocketOptions> webSocketConfiguration = null)
+            Action<ClientWebSocketOptions> webSocketConfiguration = null, string sessionId = null)
```

Comment:
Any reason why this isn't of type `Guid?` instead of a `string`? [It looks](http://tinkerpop.apache.org/docs/3.4.6/dev/io/#_session_eval_3) to me like this should be a `Guid`:

```js
"session" : {
  "@type" : "g:UUID",
  "@value" : "41d2e28a-20a4-4ab0-b379-d810dede3786"
}
```

Reply:
I checked sessionId in document, here is `Guid`, but String request [in code](https://github.com/apache/tinkerpop/blob/master/gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/op/session/SessionOpProcessor.java#L210), and it fails when `Guid` in args

```
[WARN] OpExecutorHandler - java.util.UUID cannot be cast to java.lang.String
java.lang.ClassCastException: java.util.UUID cannot be cast to java.lang.String
	at org.apache.tinkerpop.gremlin.server.op.session.SessionOpProcessor.getSession(SessionOpProcessor.java:203)
```
anything, we should update document or add `UUID` support in `gremlin-server`

Reply:
That's strange. @spmallette, can you provide some insight on why this doesn't work with a sessionId as `UUID`/`GUID` although it's documented like that?

Reply:
The session id is a string but you often see a UUID string for convenience.

Reply:
But it's serialized as `g:UUID` in GraphSON in our docs [here](http://tinkerpop.apache.org/docs/3.4.6/dev/io/#_session_eval_3). Is that an error in the docs or am I missing something here?
But OK, that means right now for this PR that the type used is correct.

Reply:
that must be an error i the docs...unless there was a time when we accepted `String` or `UUID` types, but i don't think we did. it looks like our IO test files are bad too because those docs basically come from those files. since it's more than just a documentation change, i will go and correct all that. thanks for pointing this issue out.